lib/pure/memfiles.nim

```diff
@@ -39,8 +46,19 @@ type
 
 {.deprecated: [TMemFile: MemFile].}
 
+proc checkConstructed(m: var MemFile) =
+  if (not m.initialized):
```

Comment:
So you use the initialized field AND the handle != INVALID_HANDLE check? Why? Seems convoluted.